### PR TITLE
Adds linter github action and fixes linting

### PR DESCRIPTION
**Pull request by FlxB2**
Uses tslint rules from [Winery](https://github.com/OpenTOSCA/winery) and converts them to eslint, because tslint is [deprecated](https://www.npmjs.com/package/tslint). Furthermore a Github action is built which runs the linting process. Lastly all current linting errors are fixed automatically.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Lint
+
+on: push
```
Comment (Legion2) [inline .github/workflows/lint.yml:3]:
```suggestion
on: [push, pull_request]
```
also trigger lint on pull requests

**Comment by T-Haller** [comment]
I get an end of line error with this config. It is fixed for me by changing line 38 in the .eslintrc.js file from `    "prettier/prettier": ["error"],` to
`"prettier/prettier": ["error", {
  "endOfLine":"auto"
}],`

**Comment by FlxB2** [comment]
> I get an end of line error with this config. It is fixed for me by changing line 38 in the .eslintrc.js file from ` "prettier/prettier": ["error"],` to
> `"prettier/prettier": ["error", { "endOfLine":"auto" }],`

Did eslint report the error, IntelliJ or Angular CLI ?

**Comment by T-Haller** [comment]
> 
> 
> > I get an end of line error with this config. It is fixed for me by changing line 38 in the .eslintrc.js file from ` "prettier/prettier": ["error"],` to
> > `"prettier/prettier": ["error", { "endOfLine":"auto" }],`
> 
> Did eslint report the error, IntelliJ or Angular CLI ?

It came from Eslint. The complete Message is "ESLint: Delete `␍`(prettier/prettier)". Also I was using WebStorm.

**Comment by Legion2** [comment]
looks like a windows line ending problem, do you use windows?

**Comment by T-Haller** [comment]
> 
> 
> looks like a windows line ending problem, do you use windows?

Yes this occurs on Windows. Haven't tried it on Linux yet.

**Comment by FlxB2** [comment]
> > looks like a windows line ending problem, do you use windows?
> 
> Yes this occurs on Windows. Haven't tried it on Linux yet.

you have to set Webstorm to use Line Feed returns (LF), you can find the option on the bottom right. In your current configuration CRLF or CR might be selected

**Comment by Legion2** [comment]
you must tell git to checkout LF, else you will have this error always after git pull. Or you change the linting to use CRLF on windows

**Comment by T-Haller** [comment]
> 
> 
> > > looks like a windows line ending problem, do you use windows?
> > 
> > 
> > Yes this occurs on Windows. Haven't tried it on Linux yet.
> 
> you have to set Webstorm to use Line Feed returns (LF), you can find the option on the bottom right. In your current configuration CRLF or CR might be selected

> 
> 
> you must tell git to checkout LF, else you will have this error always after git pull. Or you change the linting to use CRLF on windows

I fixed it locally in now, but could you add the .idea folder to the gitignore otherwise this will probably be void after the next pull

**Comment by Legion2** [comment]
If it's a windows related problem you are not the only one having it, so the lint config must be fixed.

**Pull request merged**